test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/MSBuildStaticGraphRestoreTests.cs

```diff
@@ -126,7 +127,33 @@ public void GetPackageDownloads_WhenNotExactVersion_ThrowsException(string versi
                     var _ = MSBuildStaticGraphRestore.GetPackageDownloads(project).ToList();
                 };
 
-                act.Should().Throw<ArgumentException>().WithMessage($"'{expected ?? VersionRange.Parse(version).OriginalString}' is not an exact version like '[1.0.0]'. Only exact versions are allowed with PackageDownload.");
+                act.Should().Throw<ArgumentException>().WithMessage($"Package 'PackageA {expected ?? VersionRange.Parse(version).OriginalString}' does not have an exact version like '[1.0.0]'. Only exact versions are allowed with PackageDownload.");
```

Comment:
```suggestion
                act.Should().Throw<ArgumentException>().WithMessage($"Package 'PackageA {expected ?? version}' does not have an exact version like '[1.0.0]'. Only exact versions are allowed with PackageDownload.");
```

I think the parsing is redundant if you'll use OriginalString.
It's supposed to give you version anyways.